### PR TITLE
Package summary with URL in quotes is parsed incorrectly

### LINKED ISSUE #5086: Package summary with URL in quotes is parsed incorrectly
For example, this package: https://www.nuget.org/packages/Microsoft.AspNet.WebHooks.Receivers.Paypal/1.2.1

Is built from this NUSPEC:

https://github.com/aspnet/WebHooks/blob/cfb4fe63ae789c602b63bc1f8f6b9a018f044a59/src/Microsoft.AspNet.WebHooks.Receivers.Paypal/Microsoft.AspNet.WebHooks.Receivers.Paypal.nuspec#L13-L15

That has this description:

```xml
    <description>
      This package provides support for receiving WebHooks from Paypal using the Paypal .NET SDK, see "https://www.nuget.org/packages/PayPal". For information about Paypal WebHooks, see "https://developer.paypal.com/webapps/developer/docs/integration/direct/rest-webhooks-overview/".
    </description>
```

When seen in the gallery, the URLs are made to be clickable, but the ending double-quote (`"`) ends up being part of the URL, which is incorrect.

Perhaps the URL parser could take this into account and not include quotes as part of the URL.

### PR DESCRIPTION
**Pull request by xavierdecoster**
Fixes #5086

### CODE REVIEW

File: src/NuGetGallery/Helpers/HtmlExtensions.cs [hunk 1]
```diff
@@ -58,10 +58,22 @@ public static IHtmlString PreFormattedText(this HtmlHelper self, string text)
 
             // Turn HTTP and HTTPS URLs into links.
             // Source: https://stackoverflow.com/a/4750468
+            MatchEvaluator anchorEvaluator = match =>
+            {
+                if (match.Value.EndsWith("&quot;"))
```
Comment (agr) [inline src/NuGetGallery/Helpers/HtmlExtensions.cs]: Make `"&quot;"` a constant maybe?

**Comment by agr** [comment]
Unit tests for an issue? Does it reproduce with other characters? `'` for example.

**Comment by loic-sharma** [comment]
To add to @agr's point, also consider examples like:

> Blah blah https://nuget.org.
> Blah blah, https://nuget.org, blah blah.
> Blah blah https://nuget.org; blah blah
> Blah blah https://nuget.org/"link-with-quotations-in-path"
> Blah blah https://nuget.org/'link-with-quotations-in-path'

File: src/NuGetGallery/Helpers/HtmlExtensions.cs (hunk 1 above)
Comment (xavierdecoster) [reply src/NuGetGallery/Helpers/HtmlExtensions.cs]: Fixed

**Comment by xavierdecoster** [comment]
@agr @loic-sharma I think there's a ton of edge cases not covered today, but the cases for single and double quotes should capture the bulk of it. Updated tests with additional cases.

This quick-fix will cover those issues that have been reported. Others may be added as the need arises?

File: tests/NuGetGallery.Facts/Helpers/HtmlExtensionsFacts.cs [hunk 2]
```diff
@@ -69,6 +69,9 @@ public void EncodesHtml()
             [InlineData("My site is http://www.nuget.org/?foo[]=a", "My site is <a href=\"http://www.nuget.org/?foo\" rel=\"nofollow\">http://www.nuget.org/?foo</a>[]=a")]
             [InlineData("http://a.com http://b.com", "<a href=\"http://a.com\" rel=\"nofollow\">http://a.com</a> <a href=\"http://b.com\" rel=\"nofollow\">http://b.com</a>")]
             [InlineData("http://www.nuget.org/ is my site.", "<a href=\"http://www.nuget.org/\" rel=\"nofollow\">http://www.nuget.org/</a> is my site.")]
+            [InlineData("\"http://www.nuget.org/\" is my site.", "&quot;<a href=\"http://www.nuget.org/\" rel=\"nofollow\">http://www.nuget.org/</a>&quot; is my site.")]
+            [InlineData("\'http://www.nuget.org/\' is my site.", "&#39;<a href=\"http://www.nuget.org/\" rel=\"nofollow\">http://www.nuget.org/</a>&#39; is my site.")]
+            [InlineData("http://www.nuget.org, is my site.", "<a href=\"http://www.nuget.org\" rel=\"nofollow\">http://www.nuget.org</a>, is my site.")]
```
Comment (agr) [inline tests/NuGetGallery.Facts/Helpers/HtmlExtensionsFacts.cs]: I'd guess the test case with brackets is also quite common:

> my package (https://www.nuget.org/api/v2/package/Newtonsoft.Json/13.0.2) is not downloadable from nuget.org

Not sure if it is properly processed by parser.

Comment (loic-sharma) [reply tests/NuGetGallery.Facts/Helpers/HtmlExtensionsFacts.cs]: Also, urls that are followed by `;` and `-` do not work today. See: https://www.nuget.org/packages/Loshar.My.Package/3.0.0

Comment (xavierdecoster) [reply tests/NuGetGallery.Facts/Helpers/HtmlExtensionsFacts.cs]: Added fix + test coverage for these cases as well.
